game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java

```diff
@@ -99,16 +99,12 @@ private void setIconMap() {
                 .collect(Collectors.toList());
         maxIconCounter = Math.max(maxIconCounter, icons.size());
         playerMap.put(playerNode.getPlayerName().getValue(), players);
-        if (uiContext == null) {
-          iconMap.put(playerNode.getPlayerName().getValue(), null);
-        } else {
-          iconMap.put(playerNode.getPlayerName().getValue(), icons);
-        }
+        iconMap.put(playerNode.getPlayerName().getValue(), uiContext == null ? null : icons);
```

Comment:
One odd thing, if `uiContext` and icons have a relationship here that is not direct or necessarily clear. EG: what happens to `icons` when uiContext is null? It seems like there is a potential for computations and then it's all ignored later when some *other* variable is null.

Would it be yet cleaner if `icons` were assigned to `null` when `uiContext` is `null`?

Reply:
Good point, will simplify something else even 👍